host/host.go

```diff
@@ -275,16 +269,18 @@ func runDaemon(args *docopt.Args) {
 	events := state.AddListener("all")
 	go syncScheduler(cluster, hostID, events)
 
-	h := &host.Host{}
-	if h.Metadata == nil {
-		h.Metadata = make(map[string]string)
-	}
+	h := &host.Host{ID: hostID, Metadata: make(map[string]string)}
 	for _, s := range metadata {
 		kv := strings.SplitN(s, "=", 2)
 		h.Metadata[kv[0]] = kv[1]
 	}
-	h.ID = hostID
 
+	var jobStream stream.Stream
+	sh.BeforeExit(func() {
+		if jobStream != nil {
+			jobStream.Close()
```

Comment:
As I said in the commit, I found it easier to follow with the `jobStream` logic being all in one section.